botocore/client.py

```diff
@@ -460,6 +499,27 @@ def _resolve_signature_version(self, service_name, resolved):
             signature_version=resolved.get('signatureVersions'))
 
 
+def _get_configured_signature_version(service_name, client_config,
```

Comment:
Why is this an internal module level function?  Doesn't seem to be used outside the client creator class.  I'd suggest moving it into the class unless it needs to be module level.